util/metrics/prometheus/prometheus_test.go

```diff
@@ -96,10 +92,16 @@ func TestInt64Set(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			mName := strings.Replace(tc.name, " ", "_", -1) + "_int"
 			m := NewInt64(mName, "fake desc", tc.fields...)
+			var wg sync.WaitGroup
 			for _, set := range tc.sets {
 				for n, fields := range set {
-					m.Set(n, fields...)
+					wg.Add(1)
+					go func(n int64, fields []string) {
+						m.Set(n, fields...)
+						wg.Done()
+					}(n, fields)
 				}
+				wg.Wait()
 			}
 			got := m.(Valuer).Values()
```

Comment:
Ideally we also ensure concurrently accessing both `Set()` and `Values()` works (aka concurrently reading from a map while someone else is writing to it)

See https://go.dev/blog/maps#concurrency

Reply:
I considered writing a separate test for this, but I found that what I wanted to test was surprisingly small.

Ex. If I set a metric to "1", and then set it to "2" and read in parallel, I don't think I care which one the read returns; I just care that it doesn't panic, and that the final read is correct (which I test already).

This seemed like the easiest way to do reads and writes in parallel so that -race can detect if a mutex is broken. Thoughts?